src/mbgl/tile/geometry_tile_data.cpp

```diff
@@ -17,71 +17,93 @@ static double signedArea(const GeometryCoordinates& ring) {
     return sum;
 }
 
-static ClipperLib::Path toClipperPath(const GeometryCoordinates& ring) {
-    ClipperLib::Path result;
+static LinearRing<std::int32_t> toWagyuPath(const GeometryCoordinates& ring) {
+    LinearRing<std::int32_t> result;
     result.reserve(ring.size());
     for (const auto& p : ring) {
         result.emplace_back(p.x, p.y);
     }
     return result;
 }
 
-static GeometryCoordinates fromClipperPath(const ClipperLib::Path& path) {
-    GeometryCoordinates result;
-    result.reserve(path.size() + 1);
-
-    for (const auto& p : path) {
-        using Coordinate = GeometryCoordinates::coordinate_type;
-        assert(p.x >= std::numeric_limits<Coordinate>::min());
-        assert(p.x <= std::numeric_limits<Coordinate>::max());
-        assert(p.y >= std::numeric_limits<Coordinate>::min());
-        assert(p.y <= std::numeric_limits<Coordinate>::max());
-        result.emplace_back(Coordinate(p.x), Coordinate(p.y));
-    }
-
-    // Clipper does not repeat initial point, but our geometry model requires it.
-    if (!result.empty()) {
-        result.push_back(result.front());
-    }
-
-    return result;
+static void pushWagyuRing(GeometryCollection & solution,
+                          mapbox::geometry::wagyu::ring_ptr<std::int32_t>& r) {
+    GeometryCoordinates lr;
+    lr.reserve(r->size() + 1);
+    auto firstPt = r->points;
+    auto ptIt = r->points;
+    do {
+        lr.emplace_back(ptIt->x, ptIt->y);
+        ptIt = ptIt->prev;
+    } while (ptIt != firstPt);
+    lr.emplace_back(firstPt->x, firstPt->y); // close the ring
+    solution.push_back(lr);
 }
 
-static void processPolynodeBranch(ClipperLib::PolyNode* polynode, GeometryCollection& rings) {
-    // Exterior ring.
-    rings.push_back(fromClipperPath(polynode->Contour));
-    assert(signedArea(rings.back()) > 0);
-
-    // Interior rings.
-    for (auto * ring : polynode->Childs) {
-        rings.push_back(fromClipperPath(ring->Contour));
-        assert(signedArea(rings.back()) < 0);
-    }
-
-    // PolyNodes may be nested in the case of a polygon inside a hole.
-    for (auto * ring : polynode->Childs) {
-        for (auto * subRing : ring->Childs) {
-            processPolynodeBranch(subRing, rings);
+static void buildWagyuResults(GeometryCollection & solution,
+                              mapbox::geometry::wagyu::ring_vector<std::int32_t>& rings) {
+    for (auto& r : rings) {
+        if (r == nullptr) {
+            continue;
+        }
+        assert(r->points);
+        if (r->size() < 3) {
+            continue;
+        }
+        solution.emplace_back();
+        pushWagyuRing(solution, r);
+        for (auto& c : r->children) {
+            if (c == nullptr) {
```

Comment:
Rather then attempting to erase items out of the middle of a vector, children often just turns the pointer to a nullptr. This was tested and found to be faster then actually erasing the pointer. Additionally if a new child is added it will overwrite a nullptr rather then adding onto the end.